samples/subsys/display/lvgl/src/main.c

```diff
@@ -59,33 +71,30 @@ int main(void)
 			return 0;
 		}
 
-		gpio_init_callback(&button_callback, button_isr_callback,
-				   BIT(button_gpio.pin));
+		gpio_init_callback(&button_callback, button_isr_callback, BIT(button_gpio.pin));
 
 		err = gpio_add_callback(button_gpio.port, &button_callback);
 		if (err) {
 			LOG_ERR("failed to add button callback: %d", err);
 			return 0;
 		}
 
-		err = gpio_pin_interrupt_configure_dt(&button_gpio,
-						      GPIO_INT_EDGE_TO_ACTIVE);
+		err = gpio_pin_interrupt_configure_dt(&button_gpio, GPIO_INT_EDGE_TO_ACTIVE);
 		if (err) {
 			LOG_ERR("failed to enable button callback: %d", err);
 			return 0;
 		}
 	}
 #endif
 
-	if (IS_ENABLED(CONFIG_LV_Z_POINTER_KSCAN) || IS_ENABLED(CONFIG_LV_Z_POINTER_INPUT)) {
-		lv_obj_t *hello_world_button;
-
-		hello_world_button = lv_btn_create(lv_scr_act());
-		lv_obj_align(hello_world_button, LV_ALIGN_CENTER, 0, 0);
-		hello_world_label = lv_label_create(hello_world_button);
-	} else {
-		hello_world_label = lv_label_create(lv_scr_act());
-	}
+#if IS_LV_POINTER_ENABLED
+	hello_world_button = lv_btn_create(lv_scr_act());
+	lv_obj_align(hello_world_button, LV_ALIGN_CENTER, 0, 0);
+	lv_obj_add_event_cb(hello_world_button, button_click_callback, LV_EVENT_CLICKED, NULL);
+	hello_world_label = lv_label_create(hello_world_button);
+#else
+	hello_world_label = lv_label_create(lv_scr_act());
```

Comment:
does this line work? `hello_world_label` only gets declared if `IS_LV_POINTER_ENABLED`. That by the ways is the reason why `if (IS_ENABLED(CONFIG_X))` is preferred to `#if CONFIG_X`. I suggest going back to the using the "if" if possible.

Reply:
Ah! my bad -- misread button/label. Anyway, any chance this could still use `if (IS_LV_POINTER_ENABLED)`?

Reply:
reverted